Core/src/org/sleuthkit/autopsy/contentviewers/WindowsRegistryViewer.java

```diff
@@ -0,0 +1,149 @@
+import com.williballenthin.rejistry.RegistryParseException;
+import java.awt.BorderLayout;
+import java.awt.Component;
+import java.awt.Cursor;
+import java.nio.ByteBuffer;
+import java.util.Arrays;
+import java.util.List;
+import java.util.logging.Level;
+import javax.swing.JPanel;
+import org.sleuthkit.autopsy.rejview.RejView;
+import org.sleuthkit.autopsy.coreutils.Logger;
+import org.sleuthkit.datamodel.AbstractFile;
+import org.sleuthkit.datamodel.Content;
+import org.sleuthkit.datamodel.TskCoreException;
+
+/**
+ * FileTypeViewer which displays information for Windows Registry files
+ */
+class WindowsRegistryViewer extends JPanel implements FileTypeViewer {
+
+    private static final long serialVersionUID = 1L;
+    private static final Logger logger = Logger.getLogger(WindowsRegistryViewer.class.getName());
+    private static final String[] SUPPORTED_MIMETYPES = new String[]{"application/x.windows-registry"};
+    //Registry log files which should be ignored share the same signature as Registry files but appear to have a size of 1024
+    private static final String LOG_FILE_EXTENSION = "log"; //base extension for log files
+    private RejView regview;
+    private AbstractFile lastFile;
+
+    WindowsRegistryViewer() {
+        super(new BorderLayout());
+    }
+
+    private void setDataView(Content content) {
+        if (content == null) {
+            this.resetComponent();
+            return;
+        }
+
+        this.setCursor(Cursor.getPredefinedCursor(Cursor.WAIT_CURSOR));
+
+        if (content.getSize() == 0) {
+            return;
+        }
+
+        byte[] data = new byte[(int) content.getSize()];
+
+        try {
+            content.read(data, 0x0, content.getSize());
+        } catch (TskCoreException ex) {
+            logger.log(Level.WARNING, "Failed to read file content.", ex);
+        }
+        ByteBuffer buf = ByteBuffer.wrap(data);
+
+        RegistryHive h = new RegistryHiveBuffer(buf);
+        this.regview = new RejView(h);
+        this.add(this.regview, BorderLayout.CENTER);
+
+        this.setCursor(null);
+    }
+
+    @Override
+    public Component getComponent() {
+        return this;
+    }
+
+    @Override
+    public void resetComponent() {
+        // cleanup anything
+        if (this.regview != null) {
+            this.remove(this.regview);
+            this.regview = null;
+        }
+        lastFile = null;
+    }
+
+    @Override
+    public boolean isSupported(AbstractFile file) {
+        if (file == null) {
+            return false;
+        }
+        if (file.getSize() == 0) {
+            return false;
+        }
+
+        if (file.getNameExtension().toLowerCase().startsWith(LOG_FILE_EXTENSION)) {
+            return false;
+        }
+        byte[] header = new byte[0x4000];
+
+        try {
+            file.read(header, 0x0, Math.min(0x4000, file.getSize()));
+        } catch (TskCoreException ex) {
+            logger.log(Level.WARNING, "Failed to read file content", ex);
```

Comment:
I guess same question, shouldn't we exit if there was an exception while trying to read data from the file? Is there a point in trying to continue and to pass incomplete/gibberish data to other classes?

Reply:
Same answer, I have no idea if there is a point.